File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        maj = n // 2
        count = 0
        elem = 0
        for i in range(n):
            if count == 0:
                elem = nums[i]
                count += 1
            elif elem == nums[i]:
                count += 1
            else:
                count -= 1
        return elem      

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 0, ele = 0;
        for (int num: nums) {
            if (num == ele) {
                count += 1;
            }
            else if (count == 0) {
                count += 1;
                ele = num;
            } else {
                count -= 1;
            }
        }

        return ele;
    }
}